react/features/noise-suppression/components/NoiseSuppressionButton.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+
+import type { Dispatch } from 'redux';
+
+import { translate } from '../../base/i18n';
+import {
+    IconShareAudio,
+    IconStopAudioShare
+} from '../../base/icons';
+import { connect } from '../../base/redux';
+import {
+    AbstractButton,
+    type AbstractButtonProps
+} from '../../base/toolbox/components';
+import { setOverflowMenuVisible } from '../../toolbox/actions';
+import { toggleNoiseSuppression } from '../actions';
+import { isNoiseSuppressionActive } from '../functions';
+
+type Props = AbstractButtonProps & {
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Dispatch<any>,
+
+}
+
+/**
+ * Component that renders a toolbar button for toggling noise suppression.
+ */
+class NoiseSuppressionButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.noiseSuppression';
+    icon = IconShareAudio;
+    label = 'toolbar.noiseSuppression';
+    tooltip = 'toolbar.noiseSuppression';
+    toggledIcon = IconStopAudioShare;
+    toggledLabel = 'toolbar.disableNoiseSuppression';
+
+    /**
+     * Handles clicking / pressing the button.
+     *
+     * @private
+     * @returns {void}
+     */
+    _handleClick() {
+        const { dispatch } = this.props;
+
+        dispatch(toggleNoiseSuppression());
+        dispatch(setOverflowMenuVisible(false));
```

Comment:
Why do we need to do this?

Reply:
in order to hide the `more` menu after the button is clicked